onnxscript/rewriter/ort_fusions/gqa.py

```diff
@@ -354,9 +355,163 @@ def rewrite(
             _outputs=3,
         )
 
+class LongRoPeGQACausalMask(pattern.RewriteRuleClassBase):
+    def __init__(self):
+        super().__init__("LongRoPeGQACausalMask", remove_nodes=False)
+        self._mask_cache = {}
+    
+    def _get_mask_key(self, attention_mask):
```

Comment:
In general, avoid creating class methods that do not require states from self, and instead make them module-level private functions for testability and clarity.